### DISCUSSION TITLE
How to set custom ruleSets

### DISCUSSION THREAD

**Discussion by Garf1eldGit** [detekt/detekt#3764] [Q&A] [answered]
There is no [documentation](https://detekt.github.io/detekt/extensions.html) about how to set custom ruleSets.
I migrate from  1.0.0-RC16 version to 1.16.0 and has error
```

> Run failed with 4 invalid config properties.
        - Property 'autoCorrect' is misspelled or does not exist.
        - Property 'test-pattern' is misspelled or does not exist.
        - Property 'complexity>LongParameterList>threshold' is deprecated. Use 'functionThreshold' and 'constructorThreshold' instead.
        - Property 'empty-blocks>EmptyFunctionBlock>ignoreOverriddenFunctions' is deprecated. Use 'ignoreOverridden' instead.
        - Property 'formatting>NoItParamInMultilineLambda' is misspelled or does not exist.
        - Property 'naming>FunctionParameterNaming>ignoreOverriddenFunctions' is deprecated. Use 'ignoreOverridden' instead.
        - Property 'naming>MemberNameEqualsClassName>ignoreOverriddenFunction' is deprecated. Use 'ignoreOverridden' instead.
        - Property 'custom_rule_set' is misspelled or does not exist.
```

gradle

```
detekt {
    toolVersion = "1.16.0"
    input = files("src/main/java", "src/main/kotlin")
    parallel = true

    config = files(configPath)
    buildUponDefaultConfig = true
    baseline = file(baselinePath)

    disableDefaultRuleSets = true
    debug = false
    ignoreFailures = false

    basePath = projectPath
    reports {
        xml {
            enabled = true
            destination = file(xmlReportPath)
        }
        html {
            enabled = true
            destination = file(htmlReportPath)
        }
    }
}
```

**Comment by cortinico** [comment, marked as answer]
> There is no [documentation](https://detekt.github.io/detekt/extensions.html) about how to set custom ruleSets.

There is actually a paragraph called `Custom RuleSets`: https://detekt.github.io/detekt/extensions.html#custom-rulesets

>  I migrate from 1.0.0-RC16 version to 1.16.0 and has error

That's a long migration (there has been 16+ versions in between), so errors are expected.
What the error message is telling you is that you have a lot of properties in your config file that are not existing anymore. I suggest you copy over the [default detekt config](https://github.com/detekt/detekt/blob/main/detekt-core/src/main/resources/default-detekt-config.yml) and re-apply your customization.
